skfda/ml/regression/linear_model.py

```diff
@@ -0,0 +1,101 @@
+from sklearn.metrics import mean_squared_error
+from sklearn.base import BaseEstimator, RegressorMixin
+from skfda.representation.basis import *
+
+import numpy as np
+
+
+class LinearScalarRegression(BaseEstimator, RegressorMixin):
+
+    def __init__(self, beta, weights=None):
+        self.beta = beta
+        self.weights = weights
+
+    def fit(self, X, y=None):
+
+        y, X, beta, wt = self._argcheck(y, X)
+
+        nbeta = len(beta)
+        nsamples = X[0].nsamples
+
+        y = np.array(y).reshape((nsamples, 1))
+
+        for j in range(nbeta):
+            xcoef = X[j].coefficients
+            xbasis = X[j].basis
+            Jpsithetaj = xbasis.inner_product(beta[j])
+            Zmat = xcoef @ Jpsithetaj if j == 0 else np.concatenate(
+                (Zmat, xcoef @ Jpsithetaj), axis=1)
+
+        if any(w != 1 for w in wt):
+            rtwt = np.sqrt(wt)
+            Zmat = Zmat * rtwt
+            y = y * rtwt
+
+        Cmat = np.transpose(Zmat) @ Zmat
+        Dmat = np.transpose(Zmat) @ y
+
+        Cmatinv = np.linalg.inv(Cmat)
+        betacoef = Cmatinv @ Dmat
+
+        mj2 = 0
+        for j in range(0, nbeta):
+            mj1 = mj2
+            mj2 = mj2 + beta[j].nbasis
+            beta[j] = FDataBasis(beta[j], np.transpose(betacoef[mj1:mj2]))
+
+        self.beta = beta
+
+    def predict(self, X):
+        return [sum(self.beta[i].inner_product(X[i][j])[0, 0] for i in
+                    range(len(self.beta))) for j in range(X[0].nsamples)]
+
+    def mean_squared_error(self, y_actual, y_predicted):
+        return np.sqrt(mean_squared_error(y_actual, y_predicted))
```

Comment:
This is the square root of the mean squared error (RMSD), so name it differently. In fact why are you using the square root?